frontend/src/_components/CreateApp.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { toast } from 'react-hot-toast';
+import Modal from '../HomePage/Modal';
+import { ButtonSolid } from '@/_ui/AppButton/AppButton';
+import { appService } from '@/_services';
+import _ from 'lodash';
+
+export function CreateApp({ closeModal, createApp, show }) {
+  const [appName, setappName] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleCreateApp = async () => {
+    if (!errorText) {
+      setIsLoading(true);
+      const appExists = await checkIfAppExists(appName);
+      if (appExists) {
+        setErrorText('App name already exists');
+      } else {
+        createApp(appName);
+        closeModal();
+      }
+      setIsLoading(false);
+    }
+  };
+
+  const checkIfAppExists = async (appName) => {
+    try {
+      const response = await appService.getAll(1, null, appName);
+      const apps = response.apps;
+      return apps.some((app) => app.name === appName);
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  };
+
+  const handleInputChange = (e) => {
+    const newAppName = e.target.value.trim();
+    const error = validateAppName(newAppName, 'App Name', true, false);
+    setErrorText(error?.errorMsg || '');
+    setappName(newAppName);
+  };
+
+  const validateAppName = (name, nameType, showError = false, allowSpecialChars = true) => {
```

Comment:
I think we have a name-checking utility fn in the file `utils.js` . please check if it possible to reuse